src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -553,13 +560,18 @@ impl Net {
         }
 
         let _metric = net_metrics.tap_write_agg.record_latency_metrics();
-        match Self::write_tap(tap, frame_iovec) {
-            Ok(_) => {
+        match Self::write_tap(tap, frame_iovec, bb) {
+            Ok(written) if written == frame_iovec.len() as usize => {
                 let len = u64::from(frame_iovec.len());
                 net_metrics.tx_bytes_count.add(len);
                 net_metrics.tx_packets_count.inc();
                 net_metrics.tx_count.inc();
             }
+            Ok(how_many) => error!(
```

Comment:
is logging enough here? Will we spot it?

Reply:
This was only for temporary debugging, this logic is untouched by the current version of the PR